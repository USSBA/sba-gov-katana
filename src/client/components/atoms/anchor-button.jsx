import React from 'react';
import { Button } from 'react-bootstrap'

class AnchorButton extends React.Component {

  scroll() {
    let element = document.getElementById(this.props.anchor);
    element.scrollIntoView({
      block: "start",
      behavior: "smooth"
    });
  }

  render() {
    return (
      <Button block className={ this.props.className } onClick={ this.scroll.bind(this) }>
        {this.props.text}
      </Button>
    )
  }
}

export default AnchorButton;
