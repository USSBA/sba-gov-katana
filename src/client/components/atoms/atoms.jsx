import CaptionText from "./caption-text"
