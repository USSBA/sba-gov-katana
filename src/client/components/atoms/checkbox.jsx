import React from 'react';
import Checkbox from 'rc-checkbox';
import './checkbox.scss';

class CheckBox extends React.Component {

  render() {
    return (
      <div  className="rc-checkbox-container">
        <label className="rc-checkbox-label">
        <Checkbox/>
          here is the label
        </label>

      </div>


      );
  }
}

export default CheckBox;