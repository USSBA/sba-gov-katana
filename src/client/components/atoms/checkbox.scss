$money-green: #609F00;

.rc-checkbox {
    white-space: nowrap;
    cursor: pointer;
    outline: none;
    display: inline-block;
    position: relative;
    width: 20px;
    height: 18px;
    left: -35px;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
}
//.rc-checkbox:hover .rc-checkbox-inner,
//.rc-checkbox-input:focus + .rc-checkbox-inner {
//    border-color: $money-green;
//}



.rc-checkbox-inner {
    position: relative;
    top: 0;
    left: 0;
    display: inline-block;
    width: 19px;
    height: 18px;
    border-width: 1px;
    border-style: solid;
    border-radius: 0px;
    border-color: #d9d9d9;
    background-color: #ffffff;
    transition: border-color 0.3s cubic-bezier(0.68, -0.55, 0.27, 1.55), background-color 0.3s cubic-bezier(0.68, -0.55, 0.27, 1.55);
}
.rc-checkbox-inner:after {
    transform: rotate(45deg);
    position: absolute;
    left: 5px;
    top: 2px;
    margin: auto;
    width: 7px;
    height: 10px;
    border: 3px solid #ffffff;
    border-top: 0;
    border-left: 0;
    content: ' ';
    animation-timing-function: cubic-bezier(0.68, -0.55, 0.27, 1.55);
    animation-duration: 0.3s;
    animation-name: amCheckboxOut;
}
.rc-checkbox-input {
    position: absolute;
    left: -15px;
    z-index: 9999;
    cursor: pointer;
    opacity: 0;
    top: -17px;
    bottom: 0;
    right: 0;
    width: 350px;
    height: 47px;
}
/* 选中状态 */
.rc-checkbox-checked:hover .rc-checkbox-inner {
    border-color: $money-green;
}
.rc-checkbox-checked .rc-checkbox-inner {
    border-color: $money-green;
    background-color: $money-green;
}
.rc-checkbox-checked .rc-checkbox-inner:after {
    transform: rotate(45deg);
    position: absolute;
    left: 5px;
    top: 2px;
    margin: auto;
    width: 7px;
    height: 10px;
    border: 3px solid #ffffff;
    border-top: 0;
    border-left: 0;
    content: ' ';
    animation-timing-function: cubic-bezier(0.68, -0.55, 0.27, 1.55);
    animation-duration: 0.3s;
    animation-name: amCheckboxOut;
}
.rc-checkbox-disabled.rc-checkbox-checked:hover .rc-checkbox-inner {
    border-color: #d9d9d9;
}
.rc-checkbox-disabled.rc-checkbox-checked .rc-checkbox-inner {
    background-color: #f3f3f3;
    border-color: #d9d9d9;
}
.rc-checkbox-disabled.rc-checkbox-checked .rc-checkbox-inner:after {
    animation-name: none;
    border-color: #cccccc;
}
.rc-checkbox-disabled:hover .rc-checkbox-inner {
    border-color: #d9d9d9;
}
.rc-checkbox-disabled .rc-checkbox-inner {
    border-color: #d9d9d9;
    background-color: #f3f3f3;
}
.rc-checkbox-disabled .rc-checkbox-inner:after {
    animation-name: none;
    border-color: #f3f3f3;
}
.rc-checkbox-disabled .rc-checkbox-inner-input {
    cursor: default;
}
@keyframes amCheckboxIn {
    0% {
        opacity: 0;
        transform-origin: 50% 50%;
        transform: scale(0, 0) rotate(45deg);
    }
    100% {
        opacity: 1;
        transform-origin: 50% 50%;
        transform: scale(1, 1) rotate(45deg);
    }
}
@keyframes amCheckboxOut {
    0% {
        opacity: 1;
    }
    100% {
        opacity: 0;
    }
}

//CUSTOM STYLES FOR CHECKBOX

.rc-checkbox-container {
    position: relative;
    width: 360px;
    height: 48px;
    border: 1px solid #C4C4C4;
    margin-bottom: 10px;
}

.rc-checkbox-container-checked {
    position: relative;
    width: 360px;
    height: 48px;
    border: 1px solid #666666;
    margin-bottom: 10px;
}

.rc-checkbox-container-focused {
    position: relative;
    width: 360px;
    height: 48px;
    border: 1px solid $money-green;
    margin-bottom: 10px;
}



.rc-checkbox-label {
    left: 53px;
    display: inline-block;
    font-size: 18px;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
}
