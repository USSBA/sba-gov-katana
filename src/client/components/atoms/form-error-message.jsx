import React from 'react';
import formHelperStyles from './form-helpers.scss';

class FormErrorMessage extends React.Component {
  render() {
    return (
      <p className={formHelperStyles.errorText}>
          {this.props.errorText}
      </p>
    );
  }
}
export default FormErrorMessage;
