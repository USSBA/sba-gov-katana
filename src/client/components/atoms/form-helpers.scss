//LINC FORM INPUTS

  @include theme-focus('money-green', '.textInput', $money-green);
  @include theme-focus('sba-blue', '.textInput', $sba-blue);
  @include theme-focus('cobalt-blue', '.textInput', $cobalt-blue);
  @include theme-focus('byzantine', '.textInput', $byzantine);

  @include theme-focus('money-green', '.textArea', $money-green);
  @include theme-focus('sba-blue', '.textArea', $sba-blue);
  @include theme-focus('cobalt-blue', '.textArea', $cobalt-blue);
  @include theme-focus('byzantine', '.textArea', $byzantine);

//COMMON INPUT STYLES

.inputContainer {
  margin-bottom: 20px !important
}

//validation text
.errorText {
    font-size: 16px;
    font-weight: 600;
    color: $error;
    margin-bottom:0;
    margin-top:0;
}

//TEXT INPUT
.textInputContainer {
    position: relative;
    margin-bottom: 0px !important;
}

.textInput {
  margin-bottom: 0px !important;
}

.textInputInvalid {
    composes: textInput;
    border: 1px solid $error !important;
    box-shadow:inset 0px 0px 0px 1px $error !important;
}

.textInputIconValid {
    font-size: 22px !important;
    color: #609F00;
    position: absolute;
    top: 50%;
    right: 15px;
    transform: translateY(-50%);
}

//TEXT AREA
.textAreaContainer {
    position: relative;
}

.textArea {
  height: 120px;
  padding-right: 37px;
}

.textAreaInvalid {
  composes: textArea;
  border: 1px solid $error !important;
  box-shadow:inset 0px 0px 0px 1px $error;
}

.textAreaIconValid {
  font-size: 22px !important;
  color: #609F00;
  position: absolute;
  top: 12px;
  right:20px;
}

.textAreaCounter{
    display: block;
    text-align: right;
    margin-right: 2px;
    font-size: 14px;
    font-style: italic;
    line-height: 18px;
    color: #666666;
}
