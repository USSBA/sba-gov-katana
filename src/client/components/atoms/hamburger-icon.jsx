import React from 'react';
import styles from "./hamburger-icon.scss";

class HamburgerIcon extends React.Component {

  getOpen() {
    return (
      <div>
        <svg className={styles.menuIconCloseNew} width="27px" height="27px" viewBox="0 0 27 27" version="1.1">
          <title>Close Button</title>
          <desc>Created with Sketch.</desc>
          <defs/>
          <g id="Style-Guide" stroke="none" strokeWidth={1} fill="none" fillRule="evenodd">
            <g id="Close-Button" fill="#333333">
              <g id="Group">
                <rect className={styles.closeIcon} id="Rectangle-7" transform="translate(13.081475, 13.081475) rotate(45.000000) translate(-13.081475, -13.081475) " x="11.5814755" y="-3.91852455" width={3} height={34} rx={2}/>
                <rect className={styles.closeIcon} id="Rectangle-7" transform="translate(13.081475, 13.081475) scale(-1, 1) rotate(45.000000) translate(-13.081475, -13.081475) " x="11.5814755" y="-3.91852455" width={3} height={34} rx={1}/>
              </g>
            </g>
          </g>
        </svg>
      </div>
    )
  }

  getClosed() {
    return (
      <div>
        <svg className={styles.menuIconHamburgerNew} width="15px" height="13px" viewBox="0 0 15 13">
          <title>hamburger icon</title>
          <desc>Created with Sketch.</desc>
          <defs/>
          <g id="Homepage" stroke="none" strokeWidth={1} fill="none" fillRule="evenodd">
            <g id="alert-mobile" transform="translate(-337.000000, -128.000000)" fill="#0B97DD">
              <g id="Rectangle-14-+-whitepaper-Copy-11" transform="translate(274.000000, 116.000000)">
                <path className={styles.hamburgerIcon} d="M78,22.4318182 C78,22.0962358 77.7167969,21.8181818 77.375,21.8181818 L63.625,21.8181818 C63.2832031,21.8181818 63,22.0962358 63,22.4318182 L63,23.6590909 C63,23.9946733 63.2832031,24.2727273 63.625,24.2727273 L77.375,24.2727273 C77.7167969,24.2727273 78,23.9946733 78,23.6590909 L78,22.4318182 Z M78,17.5227273 C78,17.1871449 77.7167969,16.9090909 77.375,16.9090909 L63.625,16.9090909 C63.2832031,16.9090909 63,17.1871449 63,17.5227273 L63,18.75 C63,19.0855824 63.2832031,19.3636364 63.625,19.3636364 L77.375,19.3636364 C77.7167969,19.3636364 78,19.0855824 78,18.75 L78,17.5227273 Z M78,12.6136364 C78,12.278054 77.7167969,12 77.375,12 L63.625,12 C63.2832031,12 63,12.278054 63,12.6136364 L63,13.8409091 C63,14.1764915 63.2832031,14.4545455 63.625,14.4545455 L77.375,14.4545455 C77.7167969,14.4545455 78,14.1764915 78,13.8409091 L78,12.6136364 Z" id="hamburger-icon"/>
              </g>
            </g>
          </g>
        </svg>
      </div>
    );

  }

  getText(open) {
    return <div className={styles.menuBtnTextNew}>{open
        ? "Close"
        : "MENU"}</div>;
  }

  render() {
    return (
      <div>
        {this.getText(this.props.isOpen)}
        {this.props.isOpen
          ? this.getOpen()
          : this.getClosed()}
      </div>
    );
  }
}

export default HamburgerIcon;
