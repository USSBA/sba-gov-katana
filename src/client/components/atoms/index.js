import ArrowButton from "./arrow-button/arrow-button.jsx";
import ButtonBase from "./button-base/button-base.jsx";
import CaptionText from "./caption-text/caption-text.jsx";
import Carousel from "./carousel/carousel.jsx";
import Checkbox from "./checkbox/checkbox.jsx";
import DecorativeDash from "./decorative-dash/decorative-dash.jsx";
import ExtraLargeTitleText from "./extra-large-title-text/extra-large-title-text.jsx";
import FormErrorMessage from "./form-error-message/form-error-message.jsx";
import HamburgerIcon from "./hamburger-icon/hamburger-icon.jsx";
import LargeGreySecondaryButton from "./large-grey-secondary-button/large-grey-secondary-button.jsx";
import LargeInversePrimaryButton from "./large-inverse-primary-button/large-inverse-primary-button.jsx";
import LargeInverseSecondaryButton from "./large-inverse-secondary-button/large-inverse-secondary-button.jsx";
import LargePrimaryButton from "./large-primary-button/large-primary-button.jsx";
import LargeSecondaryButton from "./large-secondary-button/large-secondary-button.jsx";
import MainLogo from "./main-logo/main-logo.jsx";
import Multiselect from "./multiselect/multiselect.jsx";
import NewsletterSmallPrimaryButton from "./newsletter-small-primary-button/newsletter-small-primary-button.jsx";
import PageLink from "./page-link/page-link.jsx";
import PdfIcon from "./icons/pdf.jsx";
import ProgressBar from "./progress-bar/progress-bar.jsx";
import Radio from "./radio/radio.jsx";
import SectionLink from "./section-link/section-link.jsx";
import SimpleSelect from "./simple-select/simple-select.jsx";
import SmallGreySecondaryButton from "./small-grey-secondary-button/small-grey-secondary-button.jsx";
import SmallGreySecondaryFormButton from "./small-grey-secondary-form-button/small-grey-secondary-form-button.jsx";
import SmallIcon from "./small-icon/small-icon.jsx";
import SmallInversePrimaryButton from "./small-inverse-primary-button/small-inverse-primary-button.jsx";
import SmallInverseSecondaryButton from "./small-inverse-secondary-button/small-inverse-secondary-button.jsx";
import SmallPrimaryButton from "./small-primary-button/small-primary-button.jsx";
import SmallPrimaryFormButton from "./small-primary-form-button/small-primary-form-button.jsx";
import SmallSecondaryButton from "./small-secondary-button/small-secondary-button.jsx";
import SocialMediaLink from "./social-media-link/social-media-link.jsx";
import SubtitleText from "./subtitle-text/subtitle-text.jsx";
import TextInput from "./text-input/text-input.jsx";
import TextArea from "./textarea/textarea.jsx";
import UtilityLink from "./utility-link/utility-link.jsx";
import DocumentType from "./document-type/document-type.jsx";
import ValidationIcon from "./validation-icon/validation-icon.jsx";
import SearchIcon from "./icons/search.jsx";
import ApplyButton from "./apply-button/apply-button.jsx";
import CaretIcon from "./icons/caret-down.jsx";
import CloseIcon from "./icons/close.jsx";
import HomeIcon from "./icons/home.jsx";
import FailureIcon from "./icons/failure.jsx";
import SuccessIcon from "./icons/success.jsx";

export { ArrowButton, ButtonBase, CaptionText, Carousel, Checkbox, DecorativeDash, ExtraLargeTitleText, FormErrorMessage, HamburgerIcon, LargeGreySecondaryButton, LargeInversePrimaryButton, LargeInverseSecondaryButton, LargePrimaryButton, LargeSecondaryButton, MainLogo, Multiselect, NewsletterSmallPrimaryButton, PageLink, PdfIcon, ProgressBar, Radio, SectionLink, SimpleSelect, SmallGreySecondaryButton, SmallIcon, SmallInversePrimaryButton, SmallInverseSecondaryButton, SmallPrimaryButton, SmallSecondaryButton, SocialMediaLink, SubtitleText, TextInput, TextArea, UtilityLink, ValidationIcon, SearchIcon, ApplyButton, DocumentType, CaretIcon, CloseIcon, HomeIcon, FailureIcon, SuccessIcon };
