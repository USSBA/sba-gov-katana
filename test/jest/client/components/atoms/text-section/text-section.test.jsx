/*global expect*/

import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, render } from 'react-testing-library'
import { TextSection } from 'atoms'

jest.mock('client/services/client-config.js', function() {
  return {
    googleAnalytics: {
      enabled: false
    }
  }
})

describe('TextSection', () => {
  test('should render a simple paragraph', () => {
    const text = '<p>test</p>'
    const component = renderer.create(<TextSection text={text} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render several paragraphs', () => {
    const text = '<p>test</p><p>test</p><p>test</p>'
    const component = renderer.create(<TextSection text={text} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render several paragraphs with an unordered list in the middle', () => {
    const text = '<p>test</p><ul><li>1</li>1<li></li></ul><p>test</p>'
    const component = renderer.create(<TextSection text={text} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  afterEach(cleanup)

  describe('.parseTables(text)', () => {
    test('should add class of "text-section-table" to the table tag', () => {
      const text = '<table></table>'
      const tableClass = 'text-section-table'
      const { container } = render(<TextSection text={text} />)
      const table = container.querySelector(`table.${tableClass}`)
      expect(table).not.toBe(null)
    })

    test('should add class of "text-section-table" to each table tag when there are multiple tables', () => {
      const text = '<table></table><table></table>'
      const tableClass = 'text-section-table'

      const { container } = render(<TextSection text={text} />)
      const table1 = container.querySelector(`table:nth-of-type(1).${tableClass}`)
      const table2 = container.querySelector(`table:nth-of-type(2).${tableClass}`)
      expect(table1).not.toBe(null)
      expect(table2).not.toBe(null)
    })

    test('adds class with indexed value to a single td tag if there is no th header tag', () => {
      const text = '<table><tr><td>First Title</td><td>Second Title</td></tr></table>'
      const td1Class = 'index-0'

      const { container } = render(<TextSection text={text} />)
      const td1 = container.querySelector(`td:nth-of-type(1).${td1Class}`)
      expect(td1).not.toBe(null)
    })

    test('adds class with indexed values to multiple td tags if there is no th header tag', () => {
      const text = '<table><tr><td>First Title</td><td>Second Title</td></tr></table>'
      const td1Class = 'index-0'
      const td2Class = 'index-1'

      const { container } = render(<TextSection text={text} />)
      const td1 = container.querySelector(`td:nth-of-type(1).${td1Class}`)
      const td2 = container.querySelector(`td:nth-of-type(2).${td2Class}`)
      expect(td1).not.toBe(null)
      expect(td2).not.toBe(null)
    })

    test('reformats td tags to include div header of class "table-header-label" and div wrapper of class "table-data-wrapper"', () => {
      const text =
        '<table><thead><tr><th>Column 1 Header</th></tr></thead><tbody><tr><td>Test Title</td></tr></tbody></table>'
      const headerClass = 'table-header-label'
      const headerText = 'Column 1 Header'
      const wrapperClass = 'table-data-wrapper'
      const wrapperText = 'Test Title'

      const { container, getByText } = render(<TextSection text={text} />)
      const divHeader = container.querySelector(`div.${headerClass}`)
      const divWrapper = container.querySelector(`div.${wrapperClass}`)
      const divHeaderText = getByText(headerText)
      const divWrapperText = getByText(wrapperText)

      expect(divHeader).not.toBe(null)
      expect(divHeaderText).not.toBe(null)
      expect(divWrapper).not.toBe(null)
      expect(divWrapperText).not.toBe(null)
    })
  })
})
