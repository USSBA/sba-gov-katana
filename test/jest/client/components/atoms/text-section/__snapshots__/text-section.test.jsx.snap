// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSection should render a simple paragraph 1`] = `
<span>
  <div
    className="textSection"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
    data-testid="text section"
  />
  <noscript />
</span>
`;

exports[`TextSection should render several paragraphs 1`] = `
<span>
  <div
    className="textSection"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p><p>test</p><p>test</p>",
      }
    }
    data-testid="text section"
  />
  <noscript />
</span>
`;

exports[`TextSection should render several paragraphs with an unordered list in the middle 1`] = `
<span>
  <div
    className="textSection"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p><ul><li>1</li>1<li></li></ul><p>test</p>",
      }
    }
    data-testid="text section"
  />
  <noscript />
</span>
`;
